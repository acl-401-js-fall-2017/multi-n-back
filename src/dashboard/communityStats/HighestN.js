import React, { PureComponent } from 'react';
import { connect } from 'react-redux';


class HighestN extends PureComponent {

  render() {
    const { leaderboard } = this.props;

    return leaderboard ?  ( 
      <article className="community-highest-n community-leaderboard">
        <h4 className="community-leaderboard-header">Highest N-backs</h4>
        <div className="community-highest-n-leaderboard">
          {Object.keys(leaderboard).map(difficulty => (
            <section className={`community-highest-n-${difficulty} community-leaderboard-of-highest-n`}>
              <h5 className="community-leaderboard-difficulty-h5">{difficulty}</h5>
              <ul>
                {leaderboard[difficulty].map(leader => (
                  <li>
                    <span className="community-leaderboard-username">{leader.name}: </span>
                    <span className="community-leaderboard-data">{leader.maxN}</span>
                  </li>
                ))}
              </ul>
            </section>
          ))}
        </div>
      </article> 
    ) : (
      <article className="highest-n"></article> 
    );
  }
}

export default connect(
  state => ({}),
  null
)(HighestN);