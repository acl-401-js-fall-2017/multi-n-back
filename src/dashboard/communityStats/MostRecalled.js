import React, { PureComponent } from 'react';
import { connect } from 'react-redux';


class MostRecalled extends PureComponent {

  render() {
    const { leaderboard } = this.props;
    return leaderboard ?  ( 
      <article className="community-most-recalled community-leaderboard">
        <h4 className="community-leaderboard-header">Most N-backs Recalled</h4>
        <section className="community-most-recalled-leaderboard">
          <ul>
            {leaderboard.map((leader, i) => (
              <li key={i}>
                <span className="community-leaderboard-username">{leader.name}: </span>
                <span className="community-leaderboard-data">{leader.count}</span>
              </li>
            ))}
          </ul>
        </section>
      </article> 
    ) : (
      <article className="most-recalled"></article> 
    );
  }
}

export default connect(
  state => ({}),
  null
)(MostRecalled);