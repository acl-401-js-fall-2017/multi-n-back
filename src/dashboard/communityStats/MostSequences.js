import React, { PureComponent } from 'react';
import { connect } from 'react-redux';


class MostSequences extends PureComponent {

  render() {

    return (
      <article className="most-sequences">

      </article>
    );
  }
}

export default connect(
  state => ({}),
  null
)(MostSequences);