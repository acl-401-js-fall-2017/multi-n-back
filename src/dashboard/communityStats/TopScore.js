import React, { PureComponent } from 'react';
import { connect } from 'react-redux';


export class TopScore extends PureComponent {
  
  render() {
    const { leaderboard } = this.props;

    return leaderboard ?  ( 
      <article className="community-top-score community-leaderboard">
        <h4 className="community-leaderboard-header">Top Scores</h4>
        <div className="community-top-score-leaderboard">
          {Object.keys(leaderboard).map(difficulty => (
            <section key={difficulty} className={`community-top-score-${difficulty} community-leaderboard-of-top-score`}>
              <h5 className="community-leaderboard-difficulty-h5">{difficulty}</h5>
              <ul>
                {leaderboard[difficulty].map((leader, i) => (
                  <li key={i}>
                    <span className="community-leaderboard-username">{leader.name}: </span>
                    <span className="community-leaderboard-data">{leader.highestScore}</span>
                  </li>
                ))}
              </ul>
            </section>
          ))}
        </div>
      </article> 
    ) : (
      <article className="top-score"></article> 
    );
  }
}

export default connect(
  state => ({}),
  null
)(TopScore);