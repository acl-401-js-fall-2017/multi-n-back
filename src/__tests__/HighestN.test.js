import React from 'react';
import { shallow } from 'enzyme';
import { HighestN } from '../dashboard/communityStats/HighestN';

describe('HighestN', () => {
  it('shows HighestN component', () => {

    const wrapper = shallow(
      <HighestN />
    );
    expect(wrapper).toMatchSnapshot();
  });
});