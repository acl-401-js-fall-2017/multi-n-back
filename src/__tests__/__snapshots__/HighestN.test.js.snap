// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighestN shows HighestN component 1`] = `
<article
  className="highest-n"
/>
`;
